builder/processor/src/main/java/io/helidon/builder/processor/TypeContext.java

```diff
@@ -0,0 +1,489 @@
+                .map(it -> "@" + it)
+                .toList();
+    }
+
+    private static List<TypeName> prototypeImplements(Annotation annotation) {
+        return annotation.value()
+                .map(value -> {
+                    return Stream.of(value.split(","))
+                            .map(TypeName::create)
+                            .toList();
+                })
+                .orElseGet(List::of);
+    }
+
+    private static void gatherExtends(TypeInfo typeInfo, Set<TypeName> extendList,
+                                      Set<TypeName> superPrototypes,
+                                      Set<TypeName> ignoredInterfaces) {
+        // if any implemented interface is a @Blueprint, we must extend the target type as well
+        // as any implemented interface is already a Prototype, we ignore additional annotations (it is our super type)
+        List<TypeInfo> typeInfos = typeInfo.interfaceTypeInfo();
+        for (TypeInfo info : typeInfos) {
+            if (info.findAnnotation(BLUEPRINT_TYPE).isPresent()) {
+                // this is a blueprint, we must implement its built type
+                TypeName typeName = info.typeName();
+                String className = typeName.className();
+                TypeName toExtend = TypeName.builder(typeName)
+                        .className(className.substring(0, className.length() - 9))
+                        .build();
+                extendList.add(toExtend);
+                superPrototypes.add(toExtend);
+                ignoredInterfaces.add(toExtend);
+                ignoredInterfaces.add(typeName);
+            }
+            boolean gatherAll = true;
+            for (TypeInfo implementedInterface : info.interfaceTypeInfo()) {
+                if (implementedInterface.typeName().equals(PROTOTYPE_TYPE)) {
+                    extendList.add(info.typeName());
+                    // this is a prototype itself, ignore additional interfaces
+                    gatherAll = false;
+                    superPrototypes.add(info.typeName());
+                    ignoredInterfaces.add(info.typeName());
+                    ignoredInterfaces.add(TypeName.builder(info.typeName())
+                                                  .className(info.typeName().className() + "Blueprint")
+                                                  .build());
+                    // also add all super interfaces of the prototype
+                    info.interfaceTypeInfo()
+                            .stream()
+                            .map(TypeInfo::typeName)
+                            .map(TypeName::genericTypeName)
+                            .forEach(ignoredInterfaces::add);
+                    break;
+                }
+            }
+            if (gatherAll) {
+                gatherExtends(info, extendList, superPrototypes, ignoredInterfaces);
+            }
+        }
+    }
+
+    @SuppressWarnings("checkstyle:ParameterNumber") // we need all of them
```

Comment:
why not just add all of these arguments into the processing context? This is going to be a nightmare if we ever have to add/modify the args.

Reply:
Because processing context is for the whole annotation processing (every processed type has the same processing context instance).
And `TypeContext` itself is exactly there to gather all things related to a single type being processed, so we can send it around without changing signatures of all methods.

Reply:
We should consider adding another processing context aggregator for these arguments.  I see several methods taking this list.